### PR TITLE
Pull data from Code for America API for github issues

### LINKED ISSUE #179: Investigate pulling project issues from the Code for America API
https://github.com/codeforamerica/cfapi

### PR DESCRIPTION
**Pull request by wdoug**
This PR closes #179 

#### What does this PR do?

Adds an initial method for getting data from the [Code for America API](https://github.com/codeforamerica/cfapi) as an option to pull in and view issues for a project. Currently, this is very preliminary and looks pretty crappy, but it won't show up until the urls are actually added to projects, so we should be able to iterate on this without negatively effecting users.

Screenshot of what it currently looks like:
screen shot 2018-04-09 at 8 13 37 pm

### CODE REVIEW

File: server/types.graphql [hunk 1]
```diff
@@ -47,4 +47,5 @@ type Project @model {
 
   name: String! @isUnique
   description: String
+  cfapiProjectUrl: String # for linking to the Code for America API: https://github.com/codeforamerica/cfapi
```
Comment (wdoug) [inline server/types.graphql]: Let's make this unique

**Comment by mugwump303** [comment]
Seems to work well for projects with no url and one with a valid cfa api url.
One possible issue, for a project with an incorrect url, we get stuck with the spinning wheel. Maybe we it to look like a page with no url in this case.

File: src/createApolloClient.js [hunk 2]
```diff
@@ -26,9 +28,19 @@ const middlewareLink = setContext(() => {
...
 });
 
+// Create a RestLink for the REST API
+// If you are using multiple link types, restLink should go before httpLink,
+// as httpLink will swallow any calls that should be routed through rest!
+const restLink = new RestLink({
+  endpoints: {
+    cfa: 'https://api.codeforamerica.org/api/'
+  },
+  credentials: 'same-origin'
```
Comment (wdoug) [inline src/createApolloClient.js]: Wouldn't we just want to say `omit` here? We aren't sending any credentials to the Code for America API

File: src/sections/ProjectIssues.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
-import withRestData from '../utils/withRestData';
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
+// import withRestData from '../utils/withRestData';
```
Comment (wdoug) [inline src/sections/ProjectIssues.js]: ✂️

File: src/sections/ProjectIssues.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
+// import withRestData from '../utils/withRestData';
```
Comment (wdoug) [inline src/sections/ProjectIssues.js]: ✂️

File: src/sections/ProjectIssues.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  );
+};
+ProjectIssues.defaultProps = {
+  issues: []
+};
+
+const issuesQuery = gql`
+  query issues($id: String!) {
+    issues(id: $id)
+      @rest(
```
Comment (wdoug) [inline src/sections/ProjectIssues.js]: That's a neat little directive!

File: src/utils/withRestData.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
```
Comment (wdoug) [inline src/utils/withRestData.js]: We can delete this since the apollo link rest client makes this way easier

**Review by eemshi — APPROVED** [review]
👍

File: server/types.graphql [hunk 7]
```diff
@@ -47,4 +47,5 @@ type Project @model {
 
   name: String! @isUnique
   description: String
+  cfapiProjectUrl: String! # for linking to the Code for America API: https://github.com/codeforamerica/cfapi
```
Comment (wdoug) [inline server/types.graphql]: Let's make this just the id

**Comment by wdoug** [comment]
@eemshi would you mind testing this again?

**Comment by eemshi** [comment]
Looks good to me! Was able to add this project.
screen shot 2018-05-24 at 7 38 09 pm

**Comment by eemshi** [comment]
Actually there's a little quirk with this - when I first create a project it doesn't actually save the cfapiProjectId. I always have to go back into Edit mode and add it again. Any idea why @wdoug ?

**Comment by eemshi** [comment]
Pushed up a fix. Should be good to merge @wdoug

**Pull request merged**